avocado/utils/multipath.py

```diff
@@ -0,0 +1,188 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Narasimhan V <sim@linux.vnet.ibm.com>
+
+"""
+Module with multipath related utility functions
+"""
+
+import time
+from . import distro
+from . import process
+
+
+class MultipathError(Exception):
+    """
+    Multipath Error
+    """
+
+    def __init__(self, msg):
+        self.msg = msg
+
+    def __str__(self):
+        return self.msg
+
+
+def form_conf_mpath_file(blacklist="", defaults_extra=""):
+    """
+    Form a multipath configuration file, and restart multipath service.
+
+    :parm blacklist: Entry in conf file to indicate blacklist section.
+    :parm defaults_extra: Extra entry in conf file in defaults section.
+    """
+    mpath_fp = open("/etc/multipath.conf", "w")
+    mpath_fp.write("defaults {\n")
+    mpath_fp.write("    find_multipaths yes\n")
+    mpath_fp.write("    user_friendly_names yes\n")
+    if defaults_extra:
+        mpath_fp.write("    %s\n" % defaults_extra)
+    mpath_fp.write("}\n")
+    if blacklist:
+        mpath_fp.write("blacklist {\n")
+        mpath_fp.write("    %s\n" % blacklist)
+        mpath_fp.write("}\n")
+    mpath_fp.close()
+    cmd = "cat /etc/multipath.conf"
+    process.system(cmd, ignore_status=True)
+    multipath_service(operation='restart')
+
+
+def multipath_service(operation='restart'):
+    """
+    Start, Stop, Restart multipath service.
+
+    :parm operation: multipath service operation to perform.
+                     Can be one of start, stop, restart.
+
+    :raise: :class:`MultipathError`, if ``multipath is inactive``.
+    """
+    if distro.detect().name == 'Ubuntu':
+        service_name = "multipath-tools"
+    else:
+        service_name = "multipathd"
+    cmd = "service %s %s" % (service_name, operation)
```

Comment:
We have avocado.utils.service to control services, won't you think that'd be better instead of using the valid, but legacy syntax `service %s %s`?